### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by WoloxAmquinte**
## Summary
- The automation of the card "Forms" is built
- I used two ways to scroll

## Screenshot
FormsCard

## Jira Card
<https://wolox-support.atlassian.net/browse/QAT-1844?atlOrigin=eyJpIjoiMjI5ZjdmZTIyMmE2NDk3NGI0YTI2NjljZjNhZGFhOGYiLCJwIjoiaiJ9>

### CODE REVIEW

File: src/main/java/dictionary/Constants.java [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
-    public static final String WEBVIEW_MENU="WebView";
-    public static final String LOGIN_MENU="Login";
-    public static final String FORMS_MENU="Forms";
-    public static final String SWIPE_MENU="Swipe";
+    public static final String HOME_MENU = "Home";
+    public static final String WEBVIEW_MENU = "WebView";
+    public static final String LOGIN_MENU = "Login";
+    public static final String FORMS_MENU = "Forms";
+    public static final String SWIPE_MENU = "Swipe";
+
```
Comment (ffasitella) [inline src/main/java/dictionary/Constants.java]: Remove this empty line

File: src/main/java/dictionary/Constants.java [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
+    public static final String TITLE_LOGIN_MENU = "Login / Sign up Form";
+    public static final String TITLE_FORMS_MENU = "Form components";
+    public static final String TITLE_SWIPE_MENU = "Swipe horizontal";
+
```
Comment (ffasitella) [inline src/main/java/dictionary/Constants.java]: Same with this line

File: src/main/java/pages/FormsPage.java [hunk 3]
```diff
@@ -12,9 +17,90 @@ public FormsPage() {
...
+        return switchTextFormn.getText();
+    }
+
+    public void clickSwitchForm() {
+        switchFormn.click();
+    }
+
+    public void selectOptionDropDown(String text) {
+        dropDownFormn.click();
+        Optional<MobileElement> option = listOptionsDownFormn.stream().filter(element -> element.getText().equals(text)).findFirst();
```
Comment (ffasitella) [inline src/main/java/pages/FormsPage.java:76]: Is it necessary to use `stream` method?

File: src/main/java/utils/SwipeHelper.java [hunk 4]
```diff
@@ -2,18 +2,73 @@
...
+        int scrollEnd = (int) (dimension.getHeight() * 0.2);
+
+        new TouchAction((PerformsTouchActions) BrowserManagement.getDriver())
+                .press(PointOption.point(0, scrollStart))
+                .waitAction(WaitOptions.waitOptions(Duration.ofSeconds(1)))
+                .moveTo(PointOption.point(0, scrollEnd))
+                .release().perform();
+    }
+
+    public static void scrollNClick(By listItems, String Text) {
```
Comment (ffasitella) [inline src/main/java/utils/SwipeHelper.java]: Rename this method to `scrollAndClick`

File: src/main/java/utils/SwipeHelper.java [hunk 5]
```diff
@@ -2,18 +2,73 @@
...
+                        "scrollIntoView(" + option + ")"));
+    }
+
+    /**
+     * Método de https://github.com/sunilpatro1985/AppiumTest_Java_And_iOS/blob/master/src/main/java/base/Util.java
+     */
+    public static void scrollDown() {
+        Dimension dimension = BrowserManagement.getDriver().manage().window().getSize();
+        int scrollStart = (int) (dimension.getHeight() * 0.5);
+        int scrollEnd = (int) (dimension.getHeight() * 0.2);
```
Comment (ffasitella) [inline src/main/java/utils/SwipeHelper.java:35]: Was this tested in different resolutions?

File: src/main/java/utils/SwipeHelper.java [hunk 6]
```diff
@@ -2,18 +2,73 @@
...
+                }
+            }
+            if (flag)
+                break;
+            else
+                scrollDown();
+        }
+    }
+
+    public static void scrollNClick(WebElement el) {
```
Comment (ffasitella) [inline src/main/java/utils/SwipeHelper.java]: Same with this method

File: src/test/java/runners/FormsTest.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+        BrowserManagement.startServer();
+    }
+
+    @AfterClass
+    public static void finish() {
+        BrowserManagement.getDriver().closeApp();
+        BrowserManagement.getDriver().quit();
+        BrowserManagement.stopServer();
+    }
+}
```
Comment (ffasitella) [inline src/test/java/runners/FormsTest.java]: These methods will be repeated in different tests. Maybe you can make this more abstract so you don't repeat code

**Review by ffasitella — CHANGES_REQUESTED** [review]
Some issues to solve before merging. Good job!

File: src/main/java/dictionary/Constants.java (hunk 2 above)
Comment (WoloxAmquinte) [reply src/main/java/dictionary/Constants.java]: Ok

File: src/main/java/dictionary/Constants.java (hunk 1 above)
Comment (WoloxAmquinte) [reply src/main/java/dictionary/Constants.java]: Ok

File: src/main/java/pages/FormsPage.java (hunk 3 above)
Comment (WoloxAmquinte) [reply src/main/java/pages/FormsPage.java:76]: A foreach with a flag can also be used. But they do practically the same

File: src/main/java/utils/SwipeHelper.java (hunk 4 above)
Comment (WoloxAmquinte) [reply src/main/java/utils/SwipeHelper.java]: Ok

File: src/main/java/utils/SwipeHelper.java (hunk 5 above)
Comment (WoloxAmquinte) [reply src/main/java/utils/SwipeHelper.java:35]: In a 5 inch device and 5.6 inch device

File: src/main/java/utils/SwipeHelper.java (hunk 6 above)
Comment (WoloxAmquinte) [reply src/main/java/utils/SwipeHelper.java]: Ok

File: src/test/java/runners/FormsTest.java (hunk 7 above)
Comment (WoloxAmquinte) [reply src/test/java/runners/FormsTest.java]: An abstract class is created for runners

**Review by ffasitella — APPROVED** [review]
Nice job! 👍🏽

**Pull request merged**